x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -249,17 +251,21 @@ private static Query translate(BinaryComparison bc, TranslatorHandler handler) {
                 isDateLiteralComparison = true;
             }
 
+            ZoneId zoneId = null;
+            if (bc.left().dataType() == DATETIME) {
```

Comment:
A handler could contain the default zoneId however since these are created statically, there's no way to pass that through during construction time.